presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/ElasticsearchTableDescription.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.elasticsearch;
+
+import com.facebook.presto.spi.ColumnMetadata;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableList;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Strings.isNullOrEmpty;
+
+/**
+ * Json description to parse a Elasticsearch table.
+ */
+public class ElasticsearchTableDescription
+{
+    private final String tableName;
+    private final String schemaName;
+    private final String hostAddress;
+    private final int port;
+    private final String clusterName;
+    private final String index;
+    private final boolean indexExactMatch;
+    private final String type;
+
+    private List<ElasticsearchColumn> columns;
+    private List<ElasticsearchColumnMetadata> columnsMetadata;
+
+    @JsonCreator
+    public ElasticsearchTableDescription(
+            @JsonProperty("tableName") String tableName,
+            @JsonProperty("schemaName") String schemaName,
+            @JsonProperty("hostAddress") String hostAddress,
+            @JsonProperty("port") int port,
+            @JsonProperty("clusterName") String clusterName,
+            @JsonProperty("index") String index,
+            @JsonProperty("indexExactMatch") boolean indexExactMatch,
+            @JsonProperty("type") String type,
+            @JsonProperty("columns") List<ElasticsearchColumn> columns)
+    {
+        checkArgument(!isNullOrEmpty(tableName), "tableName is null or is empty");
+        checkArgument(!isNullOrEmpty(hostAddress), "hostAddress is null or is empty");
+        checkArgument(!isNullOrEmpty(String.valueOf(port)), "port is null or is empty");
+        checkArgument(!isNullOrEmpty(clusterName), "clusterName is null or is empty");
+        checkArgument(!isNullOrEmpty(type), "type is null or is empty");
+        this.tableName = tableName;
+        this.schemaName = schemaName;
+        this.hostAddress = hostAddress;
+        this.port = port;
+        this.clusterName = clusterName;
+        this.index = index;
+        this.indexExactMatch = indexExactMatch;
+        this.type = type;
+        if (columns == null) {
+            this.columns = null;
```

Comment:
Why don't you use an empty list instead of a null?

Reply:
will fix